os-fpga/FOEDAG
# Pin assign/add constraint

> ### Motivate of the pull request
> - [ ] To address an existing issue. 
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
>- Add pin assignment option in Add Constraint Form dialog.

> #### What does this pull request change?
>  This PR improves in the following aspects:  
>- Pin Assignment Method Selection.
>- Tcl Interface for placement settings.
>- Pin Assignment Selection in Add Constraint Form while creating new project design.

> 
> 

> ### Testing Steps:
>- create_design project_1
>- At "Add Constraint" Dialog select either of the pin assign option.
>- Check on the placement tasks setting either the selected option is set or not.

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Added new options to GUI for Add Constraint Forms

> ### Update Screenshots
![image](https://user-images.githubusercontent.com/109368108/185117625-7d27123f-59ba-4004-b424-2db7b86ad35b.png)

![image](https://user-images.githubusercontent.com/109368108/185117765-4f0f9f33-6cdd-4733-8beb-375424114298.png)
